includes/admin/class-sensei-editor-wizard.php

```diff
@@ -88,4 +89,124 @@ public function enqueue_admin_scripts( $hook_suffix ) {
 			Sensei()->assets->enqueue( 'sensei-editor-wizard-style', 'admin/editor-wizard/style.css' );
 		}
 	}
+
+	/**
+	 * Register block patterns.
+	 *
+	 * @access private
+	 */
+	public function register_block_patterns() {
+		$post_type = $this->get_early_post_type();
+
+		if ( ! $post_type ) {
+			return;
+		}
+
+		// Register block pattern category.
+		if ( in_array( $post_type, [ 'course', 'lesson' ], true ) ) {
+			register_block_pattern_category(
+				'sensei-lms',
+				array( 'label' => __( 'Sensei LMS', 'sensei-lms' ) )
+			);
+		}
+
+		if ( 'course' === $post_type ) {
+			$this->register_course_block_patterns();
+		} elseif ( 'lesson' === $post_type ) {
+			$this->register_lesson_block_patterns();
+		}
+	}
+
+	/**
+	 * Get post type in the init hook.
+	 *
+	 * @return string|false Post type on success, false on failure.
+	 */
+	private function get_early_post_type() {
```

Comment:
I needed this method in order to identify the post type in the `init` hook because the [documentation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-patterns/#register_block_pattern) says that it should be registered in this hook.